File: backend/src/models/Service.js
const mongoose = require("mongoose");

const serviceSchema = new mongoose.Schema(
  {
    name: { type: String, required: true, trim: true },
    description: { type: String, trim: true },
    fee: { type: Number, required: true, min: 0 },
    processingDays: { type: Number, default: 1 },
    status: { type: String, enum: ["active", "inactive"], default: "active" },
  },
  { timestamps: true }
);

module.exports = mongoose.model("Service", serviceSchema);

File: frontend/admin/lib/models/service.dart
class CampusService {
  final String id;
  final String name;
  final String description;
  final String category;
  final num fee;
  final int processingDays;
  final String status;

  CampusService({
    required this.id,
    required this.name,
    required this.description,
    required this.category,
    required this.fee,
    required this.processingDays,
    required this.status,
  });

  factory CampusService.fromJson(Map<String, dynamic> json) {
    return CampusService(
      id: json["_id"] ?? "",
      name: json["name"] ?? "",
      description: json["description"] ?? "",
      category: json["category"] ?? "General",
      fee: json["fee"] ?? 0,
      processingDays: json["processingDays"] ?? 1,
      status: json["status"] ?? "active",
    );
  }
}
